ndc-reference/bin/reference/main.rs

```diff
@@ -1236,6 +1246,30 @@ fn eval_order_by_single_column_aggregate(
     eval_aggregate_function(&function, values)
 }
 // ANCHOR_END: eval_order_by_single_column_aggregate
+
+// ANCHOR: eval_column_field_path
+fn eval_column_field_path(
+    row: &Row,
+    column_name: &String,
+    field_path: &Option<Vec<String>>,
+) -> Result<serde_json::Value> {
+    match field_path {
+        None => eval_column(row, column_name),
+        Some(path) => row
```

Comment:
Let's reuse `eval_column` here for the better error message, and `?` over `and_then`, please, to be consistent.

Reply:
Done